pallets/manta-pay/src/lib.rs

```diff
@@ -1084,10 +1194,47 @@ pub enum TransferLedgerError {
     SenderLedgerError(SenderLedgerError),
 
     /// Receiver Ledger Error
-    ReceiverLedgerError(ReceiverLedgerError),
+    ReceiverLedgerError(ReceiverLedgerError<T>),
 
     /// Invalid Transfer Shape
     InvalidTransferShape,
+
+    /// Proof System Error
+    ProofSystemError(ProofSystemError<config::Config>),
+
+    /// Invalid Transfer Proof Error
+    ///
+    /// Validity of the transfer could not be proved by the ledger.
+    InvalidProof,
+
+    /// Type Marker Parameter
+    Marker(PhantomData<T>),
+}
+
+impl<T> From<TransferLedgerError<T>> for TransferPostError<T>
+where
+    T: Config,
+{
+    #[inline]
+    fn from(value: TransferLedgerError<T>) -> Self {
+        match value {
+            TransferLedgerError::InvalidProof => Self::InvalidProof,
+            TransferLedgerError::InvalidTransferShape => Self::InvalidShape,
+            TransferLedgerError::SenderLedgerError(err) => Self::Sender(err.into()),
+            TransferLedgerError::ReceiverLedgerError(err) => Self::Receiver(err.into()),
```

Comment:
I think here receiver be unknown error and not shutdown

Reply:
Added,